test/uk/gov/hmrc/gform/sharedmodel/FormComponentSpec.scala

```diff
@@ -1852,6 +1853,78 @@ class FormComponentSpec extends Spec {
       ))
   }
 
+  it should "parse as Overseas address" in {
+    val fieldValue = toFieldValue("""|{
+                                     |  "type": "overseasAddress",
+                                     |  "id": "colombiaAddress",
+                                     |  "label": "Home",
+                                     |  "line2Mandatory": "true",
+                                     |  "cityMandatory": "true",
+                                     |  "postcodeMandatory": "true",
+                                     |  "value": {
+                                     |    "line1": "Entrada 7 Via Cerritos",
```

Comment:
Instead of defining individual attributes for mandatoriness, why not create an array of string, with values referring to the "value" fields? This introduces a correlation between the mandatory definition and the value definition

```{
  "type": "overseasAddress",
  "id": "colombiaAddress",
  "label": "Home",
  "mandatoryFields": ["line1", "city", "country"], 
  "value": {
    "line1": "Entrada 7 Via",
    "line2": "Condominio Samanes de Tacurumbi",
    "line3": "Casa 2",
    "city": "Pereira",
    "postcode": "600007",
    "country": "Colombia"
  }
}```

Reply:
You cannot make `"line1", "city", "country"` optional. They must be always mandatory.

Wrt to improving this:
```
"line2Mandatory": "false",
"cityMandatory": "true",
"postcodeMandatory": "false",
```

yes, it is possible, but I think current proposal is more readable and easier to understand for people than:

```
alsoMandatory: ["city"]
```

On ADT side we don't really care what specification looks like.

Reply:
Ok, its clear now, thanks